src/fetchPost.js

```diff
@@ -0,0 +1,365 @@
+      "sec-fetch-site": "same-site",
+    },
+    referrer: "https://divar.ir/",
+    referrerPolicy: "origin",
+    body: null,
+    method: "GET",
+    mode: "cors",
+    credentials: "omit",
+  })
+    .then((r) => r.json())
+    .then((b) => {
+      console.log(b);
+      let bookmarks = JSON.parse(window.localStorage.getItem("bookmarks"));
+      let notes = JSON.parse(window.localStorage.getItem("notes"));
+
+      let bookmarked = false;
+      if (bookmarks) {
+        for (let i = 0; i < bookmarks.length; i++) {
+          if (bookmarks[i].url == url) {
+            bookmarked = true;
+          }
+        }
+      }
+
+      let data = b.sections;
+      let title = deepSearchByKey(data, "title")["title"];
+      let author = deepSearchByKey(data, "subtitle")["subtitle"];
+      let chips = deepSearchByKey(data, "chips")["chips"];
+      let breadcrumb = deepSearchByPair(data, "widget_type", "BREADCRUMB")[0][
+        "data"
+      ]["parent_items"];
+      let infobox = null;
+      try {
+        infobox = deepSearchByPair(data, "widget_type", "GROUP_INFO_ROW")[0][
+          "data"
+        ]["items"];
+      } catch (error) {}
+
+      let rows = deepSearchByPair(data, "widget_type", "UNEXPANDABLE_ROW");
+      let description = deepSearchByPair(
+        data,
+        "widget_type",
+        "DESCRIPTION_ROW"
+      )[0]["data"]["text"];
+      description = description.replaceAll("\n", "<br/>");
+
+      let hasImage =
+        deepSearchByPair(data, "widget_type", "IMAGE_CAROUSEL").length > 0;
+      let images = hasImage
+        ? deepSearchByPair(data, "widget_type", "IMAGE_CAROUSEL")[0]["data"][
+            "items"
+          ]
+        : [];
+
+      let container = document.querySelector(".single-post-container");
+      container.innerHTML = `
+      <main class="single-post">
+        <nav class="single-post__breadcrumb">
+          ${breadcrumb
+            .map((item) => {
+              return `
+          <span class="single-post__arrowed-text"> ${
+            item.title
+          } <img class="single-post__arrow"src="${
+                require("./icons/svgs/chevron-left.svg").default
+              }"/>
+          </span>`;
+            })
+            .join("")}
+        </nav>
+        <article class="single-post__main-content">
+        <section class="single-post__right-box">
+          <h1 class="single-post__title">
+            ${title}
+          </h1>
+          <section class="single-post__author">
+            ${author}
+          </section>
+          <section class="single-post__actions">
+            <button id="contact-info" class="single-post__action-button button button--primary">
+              <span class="button__title"> اطلاعات تماس </span>
+            </button>
+            <button class="single-post__action-button button button--secondary">
+              <span class="button__title"> چت </span>
+            </button>
+            <button class="single-post__action-bookmark ${
```

Comment:
You're creating a pseudo block here, you're treating `action` as a block of its own, but you know it's not independent enough to make it, so it almost looks like you're doing `single-post__action__bookmark` but it's not BEM so you're only changing the way you separate `action` and `bookmark` from each other!